aptihramy-backend/api.py

```diff
@@ -0,0 +1,88 @@
+from fastapi import FastAPI, Depends, Query, Form
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi import FastAPI, File, UploadFile, HTTPException
+from blitzbeaver.literals import ID
+from fastapi.responses import FileResponse
+from database import Database
+import os
+from database_config import get_database
+from constants import COLUMN_RAW_TO_PRETTY, COLUMN_PRETTY_TO_RAW
+from pydantic import BaseModel
+from typing import Dict, List
+import time as time
+from fastapi import FastAPI
+from fastapi.responses import JSONResponse
+from typing import Dict
+import ast
+import json
+
+app = FastAPI()
+
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+class FilterResponse(BaseModel):
+    data: Dict[ID, List[List[str]]]
+
+
+class FilterRequest(BaseModel):
+    data: Dict[str, str]
+
+
+@app.get("/")
+def read_root():
+    return {"message": "Hello, FastAPI!"}
+
+FOLDER_PATH = "pages"
+
+
+@app.get("/images/{filename}")
+async def get_image(filename: str):
+    file_path = os.path.join(FOLDER_PATH, filename)
+    if os.path.exists(file_path):
+        return FileResponse(
+            file_path, media_type="image/jpeg"
+        )  # Adjust media type as needed
+    return {"error": "File not found"}
+
+
+@app.get("/filter")
+def filter_data(
+    filters: str,
+    db: Database = Depends(get_database),
+):
+    feature_search_value = json.loads(filters)
```

Comment:
You shouldn't expect data in JSON format passed as URL argument, either take a dynamic number of URL arguments or a request body (with a POST endpoint).